src/util.rs

```diff
@@ -216,10 +216,16 @@ pub fn access_of(register: &Register) -> Access {
 }
 
 /// Turns `n` into an unsuffixed separated hex token
-pub fn hex(n: u32) -> Tokens {
+pub fn hex(n: u64) -> Tokens {
     let mut t = Tokens::new();
-    let (h2, h1) = ((n >> 16) & 0xffff, n & 0xffff);
-    t.append(if h2 != 0 {
+    let (h4, h3, h2, h1) = ((n >> 48) & 0xffff, (n >> 32) & 0xffff, (n >> 16) & 0xffff, n & 0xffff);
+    t.append(if n == 1 {
+        Ident::from("1")
```

Comment:
This special case is weird...

Reply:
What are you offering?

Reply:
Getting rid of it? I'd fully expect this function to always return a real hex literal, not a hex literal or "1".

Reply:
Or you about "1"? I can delete this.

Reply:
Yes, I don't like a function called `hex` to treat the number 1 special and not turn it into hex. ;)